staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource_test.go

```diff
@@ -0,0 +1,621 @@
+					Time:       &managedFieldTime,
+				},
+			},
+		},
+		{
+			desc: "transform all entries",
+			input: []metav1.ManagedFieldsEntry{
+				{
+					Manager:    "manager-1",
+					Operation:  metav1.ManagedFieldsOperationApply,
+					APIVersion: "apps/v1",
+					FieldsType: "FieldsV1",
+					FieldsV1:   &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:replicas":{}}}`)},
+				},
+				{
+					Manager:    "manager-2",
+					Operation:  metav1.ManagedFieldsOperationApply,
+					APIVersion: "apps/v1",
+					FieldsType: "FieldsV1",
+					FieldsV1:   &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:replicas":{}}}`)},
+				},
+			},
+			expected: []metav1.ManagedFieldsEntry{
+				{
+					Manager:    "manager-1",
+					Operation:  metav1.ManagedFieldsOperationApply,
+					APIVersion: "autoscaling/v1",
+					FieldsType: "FieldsV1",
+					FieldsV1:   &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:replicas":{}}}`)},
+				},
+				{
+					Manager:    "manager-2",
+					Operation:  metav1.ManagedFieldsOperationApply,
+					APIVersion: "autoscaling/v1",
+					FieldsType: "FieldsV1",
+					FieldsV1:   &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:replicas":{}}}`)},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		handler, err := NewScaleHandler(
+			test.input,
+			schema.GroupVersion{Group: "apps", Version: "v1"},
+			defaultMappings(),
+		)
+		if err != nil {
+			t.Fatalf("expected no error creating handler but got %v", err)
+		}
+		subresourceEntries, err := handler.ToSubresource()
+		if err != nil {
+			t.Fatalf("test %q - expected no error but got %v", test.desc, err)
+		}
+
+		if !reflect.DeepEqual(subresourceEntries, test.expected) {
+			t.Fatalf("test %q - expected output to be:\n%v\n\nbut got:\n%v", test.desc, test.expected, subresourceEntries)
+		}
+	}
+}
+
+func TestTransformingManagedFieldsToParent(t *testing.T) {
+	tests := []struct {
+		desc        string
+		parent      []metav1.ManagedFieldsEntry
+		subresource []metav1.ManagedFieldsEntry
+		expected    []metav1.ManagedFieldsEntry
+	}{
+		{
+			desc: "different-managers: apply -> update",
+			parent: []metav1.ManagedFieldsEntry{
+				{
+					Manager:    "test",
+					Operation:  metav1.ManagedFieldsOperationApply,
+					APIVersion: "apps/v1",
+					FieldsType: "FieldsV1",
+					FieldsV1:   &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:replicas":{},"f:selector":{}}}`)},
+				},
+			},
+			subresource: []metav1.ManagedFieldsEntry{
+				{
+					Manager:     "scale",
+					Operation:   metav1.ManagedFieldsOperationUpdate,
+					APIVersion:  "autoscaling/v1",
+					FieldsType:  "FieldsV1",
+					FieldsV1:    &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:replicas":{}}}`)},
+					Subresource: "scale",
+				},
+			},
+			expected: []metav1.ManagedFieldsEntry{
+				{
+					Manager:    "test",
+					Operation:  metav1.ManagedFieldsOperationApply,
+					APIVersion: "apps/v1",
+					FieldsType: "FieldsV1",
+					FieldsV1:   &metav1.FieldsV1{Raw: []byte(`{"f:spec":{"f:selector":{}}}`)},
```

Comment:
How do you know to remove the replicas field? It should only be removed if the value actually changes, no?

Reply:
Here "subresource" represent the managed fields of a Scale object _after_ it has been transformed from the old scale. 

Here is an example:
- Suppose there is a deployment with replicas set to 1, replicas is owned by manager `A`
- I apply an `autoscaling/v1` object with manager `B` and replicas set to 1 as well
- The field manager of scale will take it into account and update the managed fields of the scale object to have both `A` and `B` co-owning replicas
- We transform these managers back to the main resource. Note that this time `A` will appear as separate entry in the main resource because we added the "scale" field to the entry.

Reply:
If we are using `Update` instead of `Apply` it's the same, but we won't keep the two managers, only the one used to scale.

TLDR: it's up to the field manager of the scale object.

Reply:
In that case, shouldn't we have a test case with two managed fields from the subresource, where one matches up with an entry from the parent?

Reply:
Fair enough, let me add it.

Reply:
Added it [here](https://github.com/nodo/kubernetes/blob/scale-ownership-tracking-deployment/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource_test.go#L533)

Reply:
Yeah, that test doesn't even look at the fields value, and it assumes that the fieldmanager update did the right thing. It's merely trying to convert the change made by the fieldmanager into deployments managedfields.